### ISSUE TITLE
Upgrading to complex.js from 2.0.11 to 2.0.12 causes some unit tests to fail

### ISSUE THREAD

**Issue by josdejong** [josdejong/mathjs#2211]
The following tests fail:

```

  4554 passing (4s)
  19 pending
  5 failing

  1) dotPow
       should elevate a number to the given power:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

'NaN' !== '0'

      + expected - actual

      -NaN
      +0

      at Object.equal (tools\approx.js:31:14)
      at deepEqual (tools\approx.js:91:13)
      at Object.deepEqual (tools\approx.js:80:9)
      at Context.<anonymous> (file:///C:/Users/wjosd/projects/mathjs/test/unit-tests/function/arithmetic/dotPow.test.js:23:12)
      at processImmediate (internal/timers.js:461:21)

  2) pow
       should exponentiate a negative number to a non-integer power:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

'NaN' !== '0'

      + expected - actual

      -NaN
      +0

      at Object.equal (tools\approx.js:31:14)
      at deepEqual (tools\approx.js:91:13)
      at Object.deepEqual (tools\approx.js:80:9)
      at Context.<anonymous> (file:///C:/Users/wjosd/projects/mathjs/test/unit-tests/function/arithmetic/pow.test.js:28:12)
      at processImmediate (internal/timers.js:461:21)

  3) pow
       should exponentiate a negative bignumber to a non-integer power:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

'NaN' !== '0'

      + expected - actual

      -NaN
      +0

      at Object.equal (tools\approx.js:31:14)
      at deepEqual (tools\approx.js:91:13)
      at Object.deepEqual (tools\approx.js:80:9)
      at Context.<anonymous> (file:///C:/Users/wjosd/projects/mathjs/test/unit-tests/function/arithmetic/pow.test.js:86:12)
      at processImmediate (internal/timers.js:461:21)

  4) pow
       should handle infinite exponents:

      AssertionError [ERR_ASSERTION]: false == true
      + expected - actual

      -false
      +true

      at Context.<anonymous> (file:///C:/Users/wjosd/projects/mathjs/test/unit-tests/function/arithmetic/pow.test.js:141:5)
      at processImmediate (internal/timers.js:461:21)

  5) sqrt
       should return a Unit with a Complex value when computing the square root of a negative unit:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+ actual - expected

+ '(NaN) m / s'
- '(5i) m / s'
      + expected - actual

      -(NaN) m / s
      +(5i) m / s

      at Context.<anonymous> (file:///C:/Users/wjosd/projects/mathjs/test/unit-tests/function/arithmetic/sqrt.test.js:70:12)
      at processImmediate (internal/timers.js:461:21)
```

As far as I can see the only change in complex.js is this commit, related to `pow`: https://github.com/infusion/Complex.js/commit/97eb63b46b6ada7dec715aa4e6ec3466368403da. @infusion any idea?

**Comment by infusion** [comment]
Hey @josdejong 

could you please provide the actual information that get passed by this test? The commit was due to https://github.com/infusion/Complex.js/issues/33  so that the fix makes sense. But we're running into an edge case here it seems.

Robert

**Comment by infusion** [comment]
Okay, looked up the tests myself, I'll add those to the complex.js test suite and work it out.

**Comment by infusion** [comment]
Got it, the problem is with your case that `(-2)^1.5 = sqrt((-2)^3) ∉ R`, so that the numerical stability shortcut is wrong here. Sorry for that issue. To keep the shortcut for `(-2)^0`, as suggested in the ticket, the correct fix is to use `>` instead of `>=`.

**Comment by infusion** [comment]
I added your cases `(-2)^1.5` and `(-8)^(1/3)` to the test suite.

Before releasing the fix: I would also like to add those lines to `toString`:

```
      if (Math.abs(a) < Complex['EPSILON']) {
        a = 0;
      }

      if (Math.abs(b) < Complex['EPSILON']) {
        b = 0;
      }
```

to avoid those ugly outputs with numbers `|x|<1e-15`. I had to change two unit test cases. Does math.js rely on the complex.js `toString` so that possible test cases could break also?

**Comment by josdejong** [comment]
> 
> 
> Got it, the problem is with your case that `(-2)^1.5 = sqrt((-2)^3) ∉ R`, so that the numerical stability shortcut is wrong here. Sorry for that issue. To keep the shortcut for `(-2)^0`, as suggested in the ticket, the correct fix is to use `>` instead of `>=`.

Thanks for pinpointing this issue, though I do not yet fully understand you, is this an improvement that needs to be made in complex.js or in mathjs?

As for changing the `toString` for small values: that will not break anything in mathjs, should be ok from that point of view. However, can't it be that a user _actually_ works with very small numbers, and it would be odd to return 0?

**Comment by infusion** [comment]
Sorry, I related to the attached issue. I fixed it already in complex.js.

Regarding `toString`: Yes, but typical addition-errors lie in the range of `Xe-16`. And as almost all operations in complex.js rely on addition to some extend, I think it's fine to cap them for printing.

**Comment by josdejong** [comment]
👍

**Comment by infusion** [comment]
Okay, 2.0.13 was released.

**Comment by josdejong** [comment]
Thanks Robert, that is very fast 🚀 😎 . This indeed fixes the broken unit tests.

**Comment by josdejong** [comment]
Published now in `v9.4.0`
